scripts/tests/test_phase44d_carry_aware_family_constant_surface.py

```
@@ -0,0 +1,102 @@
+            for spec in MODULE.FAMILY_SPECS
+        }
+        self.assertEqual(rows["2x2"], {"memory": 21, "instructions": 65, "columns": 111})
+        self.assertEqual(rows["3x3"], {"memory": 28, "instructions": 82, "columns": 112})
+        self.assertEqual(rows["default"], {"memory": 37, "instructions": 103, "columns": 113})
+
+    def test_script_derives_expected_2x2_frontier_ratios_from_checked_evidence(self):
+        with tempfile.TemporaryDirectory() as tempdir:
+            temp = Path(tempdir)
+            out_json = temp / "out.json"
+            out_tsv = temp / "out.tsv"
+            completed = subprocess.run(
```

Ruff (0.15.11)

[error] `subprocess` call: check for execution of untrusted input

(S603)